test/reducedim.jl

```diff
@@ -610,9 +610,10 @@ function unordered_test_for_extrema(a; dims_test = ((), 1, 2, (1,2), 3))
     for dims in dims_test
         vext = extrema(a; dims)
         vmin, vmax = minimum(a; dims), maximum(a; dims)
-        @test isequal(extrema!(copy(vext), a), vext)
+        @test isequal(extrema!(similar(vext, NTuple{2, eltype(a)}), a), vext)
```

Comment:
This fails as was originally written because this is reducing over a `Union{T, Missing}` and depending upon the values in `a`, it `vext` might incrementally widen to a `T[]` or a `Missing[]` or a `Union{T, Missing}`... which then isn't able to be used in general.  This would be representative of a class of failure that this PR might introduce in the ecosystem.